src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -769,6 +769,10 @@ public function setDeprecated($status = true, $template = null)
             }
 
             $this->deprecationTemplate = $template;
+        } elseif (true === $status) {
```

Comment:
comparing to true is inconsistent with the cast to bool below. I'd suggest to just do `if ($status) {` here